packages/workmanager/tizen/src/workmanager_tizen_plugin.cc

```diff
@@ -0,0 +1,481 @@
+
+          bundle_free(bund);
+          return;
+        }
+      }
+      bundle_free(bund);
+
+      result->Success();
+    } else if (method_name == kCancelTaskByUniqueName) {
+      auto name =
+          GetOrNullFromEncodableMap<std::string>(&map, kCancelTaskUniqueName);
+      if (!name.has_value()) {
+        result->Error(kInvalidArg, "No name provided");
+        return;
+      }
+
+      bundle *bund = bundle_create();
+      if (!bund) {
+        LOG_ERROR("Failed create bundle");
+        result->Error(kOperationFailed, "Failed Creating bundle.");
+      }
+
+      bundle_add_str(bund, kMethodNameKey, method_name.c_str());
+      bundle_add_str(bund, kCancelTaskByUniqueName, name.value().c_str());
+
+      if (is_service_app_) {
+        TaskInfoCallback(nullptr, bund, nullptr);
+      } else {
+        int ret = event_publish_app_event(event_id.c_str(), bund);
+
+        if (ret != BUNDLE_ERROR_NONE) {
+          LOG_ERROR("Failed publish event: %s", get_error_message(ret));
+          result->Error(kOperationFailed, "Failed Publish event.");
+
+          bundle_free(bund);
+          return;
+        }
+      }
+
+      bundle_free(bund);
+
+      result->Success();
+    } else {
+      result->NotImplemented();
+    }
+  }
+
+  void HandleBackground(const FlMethodCall &call,
+                        std::unique_ptr<FlMethodResult> result) {
+    if (call.method_name() == kBackgroundChannelInitialized) {
+      is_background_initialized_ = true;
+    }
+
+    result->Success();
+  }
+
+  static void RunBackgroundCallback(const std::string &job_id,
+                                    const std::string &payload) {
+    if (!background_channel_.has_value()) {
+      LOG_ERROR("Background channel is not initialized.");
+      return;
+    }
+
+    flutter::EncodableMap arg = {
+        {flutter::EncodableValue(kBgChannelDartTask),
+         flutter::EncodableValue(job_id)},
+    };
+
+    if (payload.size() > 0) {
+      arg[flutter::EncodableValue(kBgChannelInputData)] =
+          flutter::EncodableValue(payload);
+    }
+
+    background_channel_.value()->InvokeMethod(
+        kOnResultSendMethod, std::make_unique<flutter::EncodableValue>(arg));
+  }
+
+  static void StartJobCallback(job_info_h job_info, void *user_data) {
+    char *job_id = nullptr;
+    job_info_get_job_id(job_info, &job_id);
+    auto &scheduler = JobScheduler::instance();
+    auto task_info = scheduler.LoadJobInfo(job_id).value();
+
+    RunBackgroundCallback(task_info.task_name, task_info.payload);
+  }
+
+  static void StopJobCallback(job_info_h job_info, void *user_data) {
+    // Empty
+  }
+
+  static std::optional<std::string> GetAppId() {
+    char *app_id;
+    int ret = app_manager_get_app_id(getpid(), &app_id);
+    if (ret == APP_MANAGER_ERROR_NONE) {
+      return std::string(app_id);
+    }
+
+    return std::nullopt;
+  }
+
+  static void TaskInfoCallback(const char *event_name, bundle *event_data,
```

Comment:
If an error occurs in this method, result->Error(..., ...); does not call
Is this intended?